client-ui/src/app/model/lineage-graph/leneage-graph-legend.models.ts

```diff
@@ -15,26 +15,31 @@
  */
 
 
-import { LINE_COLOR_HLT_IMPACT, LINE_COLOR_HLT_LINEAGE, LINE_COLOR_HLT_PRIMARY } from './leneage-graph.models';
+import {LINE_COLOR_HLT_IMPACT, LINE_COLOR_HLT_LINEAGE, LINE_COLOR_HLT_PRIMARY} from './leneage-graph.models';
 
 
-export interface LineageGraphLegend {
+export type LineageGraphLegend = {
   title: string;
   color: string;
   description: string;
 }
 
-export function getLineageGraphLegend(): LineageGraphLegend[] {
-  return [
-    {
+export type LineageGraphLegendKeys =
```

Comment:
It is better to use enum here like so:

```
export enum LineageGraphLegendType {
  Usage = 'usage',
  Lineage = 'lineage',
  Impact = 'impact'
}
```

and then use it in object definition and in all app to access that:

```
LineageGraphLegends[LineageGraphLegendType.Usage]
```

Reply:
IMO this way in comparison to the types approach you are missing the type safely.